# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1763327083 -3600
# Parent  1d453131973eb08a789958624afc827654a7ab09
Feature: When multiple message are selected, current message is shown. Also fix summarising for news folders (bug 1826633).

Relates to https://bugzilla.mozilla.org/show_bug.cgi?id=364896 and
           https://bugzilla.mozilla.org/show_bug.cgi?id=454829

diff --git a/mail/app/profile/all-thunderbird.js b/mail/app/profile/all-thunderbird.js
--- a/mail/app/profile/all-thunderbird.js
+++ b/mail/app/profile/all-thunderbird.js
@@ -1456,8 +1456,10 @@ pref("mail.download.open_attachments_rea
 // No reports were received for BB 128 and it appears to be
 // working again in TB 146. So remove the following line for now.
 // pref("ui.prefersReducedMotion", 0);
 
+pref("mail.multiselect_message_shows_current", false);
+
 // Unified toolbar
 
 // 0: icons beside text
 // 1: icons above text
diff --git a/mail/base/content/about3Pane.js b/mail/base/content/about3Pane.js
--- a/mail/base/content/about3Pane.js
+++ b/mail/base/content/about3Pane.js
@@ -5112,9 +5112,23 @@ var threadPane = {
         messagePane.displayMessage(uri);
         break;
       }
       default:
-        messagePane.displayMessages(gDBView.getSelectedMsgHdrs());
+        if (
+          Services.prefs.getBoolPref(
+            "mail.multiselect_message_shows_current",
+            false
+          ) ||
+          gFolder instanceof Ci.nsIMsgNewsFolder
+        ) {
+          // This will fail on a summary/dummy row.
+          try {
+            const uri = gDBView.getURIForViewIndex(threadTree.currentIndex);
+            messagePane.displayMessage(uri);
+          } catch (ex) {}
+        } else {
+          messagePane.displayMessages(gDBView.getSelectedMsgHdrs());
+        }
         break;
     }
 
     updateZoomCommands();
